enricher/api/src/main/java/io/fabric8/maven/enricher/api/AbstractHealthCheckEnricher.java

```diff
@@ -0,0 +1,90 @@
+package io.fabric8.maven.enricher.api;
+
+import java.util.List;
+
+import io.fabric8.kubernetes.api.model.Container;
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.api.model.KubernetesListBuilder;
+import io.fabric8.kubernetes.api.model.Probe;
+import io.fabric8.kubernetes.api.model.extensions.Deployment;
+
+/**
+ * Enriches containers with health check probes.
+ */
+public abstract class AbstractHealthCheckEnricher extends BaseEnricher {
+
+    public AbstractHealthCheckEnricher(EnricherContext buildContext, String name) {
+        super(buildContext, name);
+    }
+
+    @Override
+    public void addMissingResources(KubernetesListBuilder builder) {
+
+        List<HasMetadata> items = builder.getItems();
+        for (HasMetadata item : items) {
+            if (item instanceof Deployment) {
+                Deployment deployment = (Deployment) item;
+                Container container = getContainer(deployment);
+                if (container != null) {
+                    if (container.getReadinessProbe() == null) {
+                        Probe probe = getReadinessProbe();
+                        if (probe != null) {
+                            log.info("Adding readiness " + describe(probe));
+                            container.setReadinessProbe(probe);
+                        }
+                    }
+
+                    if (container.getLivenessProbe() == null) {
+                        Probe probe = getLivenessProbe();
+                        if (probe != null) {
+                            log.info("Adding liveness " + describe(probe));
+                            container.setLivenessProbe(probe);
+                        }
+                    }
+                }
+            }
+        }
+        builder.withItems(items);
+    }
+
+    private Container getContainer(Deployment deployment) {
+        if (deployment.getSpec() != null &&
+                deployment.getSpec().getTemplate() != null &&
+                deployment.getSpec().getTemplate().getSpec() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers() != null &&
+                deployment.getSpec().getTemplate().getSpec().getContainers().size() > 0) {
+
+            List<Container> containers = deployment.getSpec().getTemplate().getSpec().getContainers();
+            return containers.get(containers.size() - 1); // last one
```

Comment:
Yes, now I understand better the point. But I don't get why enrichers should be bound to generators. I mean, if I don't want to use Alpine and use a different O.S., or another JVM, I'd provide my own image but probably I still want to have health checks installed in my spring-boot or vertx application.

If the images has built-in health checks I can specify them in the configuration (or if it's a well known image, an enricher can create them). I can also deactivate health-check enrichers using profiles if I don't need them... Maybe I don't have the whole picture.

In the meantime I'll refactor to create the healthcheck module (I'm ok with it...). We can later decide what's best adding in case of user specified images. I just think that defaulting to create health checks is ok, since they can be deactivated (and the contrary is not true afaik).